drs-auth_client/lib/drs/auth_client/client.rb

```diff
@@ -0,0 +1,65 @@
+require "active_support/core_ext/hash/keys"
+require "active_support/core_ext/string/inflections"
+require "faraday"
+
+require "drs/auth_client/models/organisation"
+require "drs/auth_client/models/profile"
+
+module Drs
+  module AuthClient
+    class Client
+      def initialize(auth_token)
+        @auth_token = auth_token
+      end
+
+      def get(path)
+        response = conn.get do |request|
+          request.url path
+          request.headers['Authorization'] = "Bearer #{@auth_token}"
+        end
+
+        if response.status == 200
+          response.body
+        else
+          nil
+        end
+      end
+
+      (Models.constants - [:Base]).each do |model_name|
+        model = Models.const_get(model_name)
+        single_resource_name = model_name.to_s.underscore.downcase
+        collection_resource_name = single_resource_name.pluralize
+
+        define_method(single_resource_name) do |uid|
+          path = "#{collection_resource_name}/#{uid}"
+          get_resource(path, model, :from_hash, nil)
+        end
+
+        define_method(collection_resource_name) do
+          get_resource(collection_resource_name, model, :collection_from_hash, [])
+        end
+      end
```

Comment:
Yep, I'm happy about that too. I had it originally like that, but then refactored it to this generic way. I'll bring it back :)